.gitignore

```diff
@@ -51,9 +51,9 @@ wheels/
 .installed.cfg
 *.egg
 src/*/version.info
-src/lightning_app/
-src/lightning_fabric/
-src/pytorch_lightning/
+src/lightning_app/*
+src/lightning_fabric/*
+src/pytorch_lightning/*
```

Comment:
When I tried adding new `py.typed` files, git didn't detect them. According to [the docs](https://git-scm.com/docs/gitignore#_pattern_format):

> It is not possible to re-include a file if a parent directory of that file is excluded.

Since we exclude each parent directory above, all of the following `!` lines below have no effect! Solution? Simply exclude all subdirectories instead of the parent directory itself 😉

Reply:
I think this notation was fine, at least my IDE is happy with it

Reply:
It definitely wasn't fine for command-line git on my system. All files in those directories were completely ignored, as explained by the git docs. If you try to add a new py.typed, git does not register it.